### PR TITLE
Cleaner/easier way for user to specify Cloudwatch metric percentiles

### PR DESCRIPTION
**Pull request by feliksik**
Two changes: 

* As proposed in #591 , prepend names with 'p'; that is: metricname_p50 instead of metricname_50. 
* Make the WithPercentiles() options much nicer: WithPercentiles(0.50, 0.99) instead of WithPercentiles(cloudwatch.Percentiles{"50": 0.50, "0.99": 0.99}) 

sorry for not including these previously, I should have added [WIP] tag, as I intended to still agree on requirements, before merge.

### CODE REVIEW

**Comment by peterbourgon** [comment]
Is there a good reason to make the breaking `p` change besides consistency with other metrics?

**Comment by feliksik** [comment]
@peterbourgon : the reasons are 1) consistency and 2) clarity (that it is a percentile). 

Whether that's good enough is arguable. I can also pull it out and just include the WithPercentiles() change, if you like.

**Comment by feliksik** [comment]
I believe there is no tag released since it was included, so I guessed it didn't matter much; that's why I  [asked](https://github.com/go-kit/kit/pull/591#issuecomment-320534931)  @cam-stitt .

The extent to which you want to keep master backwards compatible is up to you.

**Comment by feliksik** [comment]
I see  that I messed up the test:  

```
--- FAIL: TestHistogram (0.23s)
	cloudwatch_test.go:182: p50: want 500.000000, have 0.000000; p90: want 532.038789, have 0.000000; p95: want 541.121341, have 0.000000; p99: want 558.158697, have 0.000000
FAIL
```

So I'll happily fix it, as soon as you can tell me which of the 2 changes you would accept :-) goodnight!

File: metrics/cloudwatch/cloudwatch.go [hunk 1]
```diff
@@ -57,16 +58,17 @@ func WithLogger(logger log.Logger) option {
...
-func WithPercentiles(p Percentiles) option {
+// WithPercentiles registers the percentiles to track, overriding the
+// existing/default values.
+func WithPercentiles(percentiles ...float64) option {
 	return func(c *CloudWatch) {
-		validated := Percentiles{}
-		for _, entry := range p {
-			if entry.f < 0 || entry.f > 1 {
-				continue // illegal entry
+		c.percentiles = make([]float64, 0)
```
Comment (seh) [inline metrics/cloudwatch/cloudwatch.go]: - [ ] Why zero here, as opposed to `len(percentiles)`?  
  It looks like you're betting in favor of them all being illegal.

Comment (feliksik) [inline metrics/cloudwatch/cloudwatch.go]: You are correct, `make([]float64, 0, len(percentiles)` would be more efficient. 

I don't care to optimize this stuff for slices that are so short; in this slice is likely to be max 4 entries. 

But I can fix it if you like.

Comment (seh) [inline metrics/cloudwatch/cloudwatch.go]: That you wrote a number at all made me take note.

I usually try to avoid the _intermediate_ reallocations (and consequent copying) in loops like these by allocating beforehand the maximum that I might need. I'd rather hang on to that unused memory (assuming it's relatively small like this) than reallocate several times accumulating the values.

I don't know how often this code executes, so it may not make any difference. For your approach, though, I recommend just removing the length entirely, deleting line 65. You can pass a nil slice argument to `append`.

Comment (feliksik) [inline metrics/cloudwatch/cloudwatch.go]: Thanks for your comment. You are right, I never needed to set it to 0. It's better to delete the line, or initialize with len(). But the current line makes no sense. 

I'll update this later :-) still waiting for a verdict on the 'p' prefix anyway.

Comment (feliksik) [inline metrics/cloudwatch/cloudwatch.go]: Hmm, thinking about it, the percentiles was already initialized by the beginning of the New() function. So not re-initializing it will add percentiles, not override the defaults. That's not good... 

For context: The method is only used once for every cloudwatch instance you make, which is generally 1 per application runtime.

Comment (seh) [inline metrics/cloudwatch/cloudwatch.go]: You _could_ write the following:
```go
c.percentiles = nil
```
Whether that's better—assuming your desire is to start here with no room—is a matter of taste.

Comment (feliksik) [inline metrics/cloudwatch/cloudwatch.go]: Seems all fine with me; I will see if I can add some tests for the custom percentiles, and will change it to either nil or Len(), whichever you like.

Comment (seh) [inline metrics/cloudwatch/cloudwatch.go]: _I_ prefer to use the length of the candidate sequence supplied to the filter—here, `len(percentiles)`—but my contribution to _go-kit_ thus far has been negligible, so I defer to any others more familiar with conventions or preferences applied more consistently throughout the project.

**Comment by feliksik** [comment]
Great, will do! As long as nobody else comments, let's do as you suggest :-)

File: metrics/cloudwatch/cloudwatch.go (hunk 1 above)
Comment (feliksik) [inline metrics/cloudwatch/cloudwatch.go]: done!

File: metrics/cloudwatch/cloudwatch_test.go [hunk 2]
```diff
@@ -193,4 +193,54 @@ func TestHistogram(t *testing.T) {
...
+		svc.mtx.RLock()
+		defer svc.mtx.RUnlock()
+		p50 = svc.valuesReceived[n50]
+		p90 = svc.valuesReceived[n90]
+
+		// our teststat.TestHistogram wants us to give p95 and p99,
+		// but with custom percentiles we don't have those.
+		// So fake them. Maybe we should make teststat.nvq() public and use that?
+		p95 = 541.121341
+		p99 = 558.158697
```
Comment (feliksik) [inline metrics/cloudwatch/cloudwatch_test.go]: @peterbourgon this may not be the nicest to hardcode; we can also make teststat.nvq() public. Do you have a preference?

Comment (peterbourgon) [inline metrics/cloudwatch/cloudwatch_test.go]: Maybe it's fine to just leave this [part of the] test out, for this particular metric?

Comment (feliksik) [inline metrics/cloudwatch/cloudwatch_test.go]: Thanks for you comment @peterbourgon ; Do you mean leave out the whole added part about testing WithPercentile() ? 
I think that's the only way. 

We can also leave it hardcoded for now; it's all fine with me. I actually do think it's useful to test the WithPercentile(), but I'm not really attached to it. 

If you have a preference, let me know; otherwise I'll leave it as-is.

**Comment by feliksik** [comment]
@peterbourgon you mentioned on slack "I rather leave it out, if you don't mind. Keep the old syntax."
Fine with me! The names are referted in the most recent commit.

**Comment by peterbourgon** [comment]
Thanks!
